aptos-move/aptos-e2e-comparison-testing/src/execution.rs

```diff
@@ -0,0 +1,358 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    check_aptos_packages_availability, compile_aptos_packages, compile_package,
+    generate_compiled_blob, is_aptos_package, DataManager, IndexReader, PackageInfo, TxnIndex,
+    APTOS_COMMONS,
+};
+use anyhow::Result;
+use aptos_framework::APTOS_PACKAGES;
+use aptos_language_e2e_tests::{data_store::FakeDataStore, executor::FakeExecutor};
+use aptos_types::{
+    contract_event::ContractEvent,
+    on_chain_config::{FeatureFlag, Features, OnChainConfig},
+    transaction::{Transaction, TransactionPayload, Version},
+    vm_status::VMStatus,
+    write_set::WriteSet,
+};
+use aptos_vm::{data_cache::AsMoveResolver, transaction_metadata::TransactionMetadata};
+use clap::ValueEnum;
+use itertools::Itertools;
+use move_core_types::{account_address::AccountAddress, language_storage::ModuleId};
+use move_package::CompilerVersion;
+use std::{collections::HashMap, path::PathBuf};
+
+fn load_packages_to_executor(
+    executor: &mut FakeExecutor,
+    package_info: &PackageInfo,
+    compiled_package_cache: &HashMap<PackageInfo, HashMap<ModuleId, Vec<u8>>>,
+) {
+    let compiled_package = compiled_package_cache.get(package_info).unwrap();
+    for (module_id, module_blob) in compiled_package {
+        executor.add_module(module_id, module_blob.clone());
+    }
+}
+
+fn load_aptos_packages_to_executor(
+    executor: &mut FakeExecutor,
+    compiled_package_map: &HashMap<PackageInfo, HashMap<ModuleId, Vec<u8>>>,
+) {
+    for package in APTOS_PACKAGES {
+        let package_info = PackageInfo {
+            address: AccountAddress::ONE,
+            package_name: package.to_string(),
+            upgrade_number: None,
+        };
+        load_packages_to_executor(executor, &package_info, compiled_package_map);
+    }
+}
+
+#[derive(ValueEnum, Clone, Copy, Debug, Eq, PartialEq, PartialOrd)]
+pub enum ExecutionMode {
+    V1,
+    V2,
+    Compare,
+}
+
+impl ExecutionMode {
+    pub fn is_v1(&self) -> bool {
+        *self == Self::V1
+    }
+
+    pub fn is_v2(&self) -> bool {
+        *self == Self::V2
+    }
+
+    pub fn is_compare(&self) -> bool {
+        *self == Self::Compare
+    }
+}
+
+impl Default for ExecutionMode {
+    fn default() -> Self {
+        Self::V1
+    }
+}
+
+pub struct Execution {
+    input_path: PathBuf,
+    execution_mode: ExecutionMode,
+    bytecode_version: u32,
+}
+
+impl Execution {
+    pub fn new(input_path: PathBuf, execution_mode: ExecutionMode) -> Self {
+        Self {
+            input_path,
+            execution_mode,
+            bytecode_version: 6,
+        }
+    }
+
+    fn set_enable(features: &mut Features, flag: FeatureFlag) {
+        let val = flag as u64;
+        let byte_index = (val / 8) as usize;
+        let bit_mask = 1 << (val % 8);
+        if byte_index < features.features.len() {
+            features.features[byte_index] |= bit_mask;
+        }
```

Comment:
deliberately ignoring features that's beyond the existing bytes? why? 

if not, why not just `features.enable(flag)`?

Reply:
I think `features.enable` was added recently. I will use this.

Reply:
done